src/Features/Core/Portable/GenerateMember/GenerateDefaultConstructors/AbstractGenerateDefaultConstructorsService.AbstractCodeAction.cs

```diff
@@ -50,19 +50,23 @@ private abstract class AbstractCodeAction : CodeAction
             }
 
             private IMethodSymbol CreateConstructorDefinition(
-                IMethodSymbol constructor)
+                IMethodSymbol baseConstructor)
             {
                 var syntaxFactory = _document.GetLanguageService<SyntaxGenerator>();
-                var baseConstructorArguments = constructor.Parameters.Length != 0
-                    ? syntaxFactory.CreateArguments(constructor.Parameters)
+                var baseConstructorArguments = baseConstructor.Parameters.Length != 0
+                    ? syntaxFactory.CreateArguments(baseConstructor.Parameters)
                     : default;
 
+                var classType = _state.ClassType;
+                var accessibility = baseConstructor.ContainingType.IsAbstractClass() ? 
+                    (classType.IsAbstractClass() ? baseConstructor.DeclaredAccessibility : Accessibility.Public)
+                    : baseConstructor.DeclaredAccessibility;       
```

Comment:
IDE style would be:

```c#
var accessibility = baseConstructor.ContainingType.IsAbstractClass()
    ? classType.IsAbstractClass() ? baseConstructor.DeclaredAccessibility : Accessibility.Public
    : baseConstructor.DeclaredAccessibility;
```

Reply:
or:

```c#
var accessibility = baseConstructor.ContainingType.IsAbstractClass()
    ? classType.IsAbstractClass()
        ? baseConstructor.DeclaredAccessibility
        : Accessibility.Public
    : baseConstructor.DeclaredAccessibility;

Reply:
interesting bit of logic... I would personally just prefer:
```c#
var accessibility  = baseConstructor.ContainingType.IsAbstractClass() && !classType.IsAbstractClass()
    ? Accessibility.Public
    : baseConstructor.DeclaredAccessibility;
```

Reply:
Done like in Neme12's suggestion